wgsl/index.bs

```diff
@@ -5480,6 +5479,74 @@ See [[#sync-builtin-functions]].
     <td>Bitwise-exclusive-or. [=Component-wise=] when |T| is a vector.
 </table>
 
+
+<table class='data'>
+  <caption>Bit shift expressions</caption>
+  <thead>
+    <tr><th>Precondition<th>Conclusion<th>Notes
+  </thead>
+
+  <tr algorithm="concrete shift left">
+    <td>|e1|: |T|<br>
+    |e2|: |TS|<br>
+    |S| is [INT]<br>
+    |T| is |S| or vec|N|&lt;|S|&gt;<br>
+    |TS| is u32 when |T| is |S|, otherwise |TS| is vec|N|&lt;u32&gt;
+    <td class="nowrap">|e1| `<<` |e2|: |T|
+    <td>Shift left (concrete):
+
+        Shift |e1| left, inserting zero bits at the least significant positions,
+        and discarding the most significant bits.
+
+        The number of bits to shift is the value of |e2|, modulo the bit width of |e1|.<br>
+
+        [=Component-wise=] when |T| is a vector.
+
+  <tr algorithm="abstract shift left">
+    <td>|e1|: |T|<br>
+    |e2|: |T|<br>
+    |T| is AbstractInt or vec|N|&lt;AbstractInt&gt;
+    <td class="nowrap">|e1| `<<` |e2|: |T|
+    <td>Shift left (abstract):
+
+        Shift |e1| left, inserting zero bits at the least significant positions,
+        and discarding the most significant bits.
+
+        The number of bits to shift is the value of |e2|.
+
+        [=Component-wise=] when |T| is a vector.
+
+        It is a [=shader-creation error=] if the calculation would overflow the AbstractInt type.
+        Overflow occurs when both 0 and 1 bits would be discarded,
+        or when the most significant bit of |e1| would differ from the most
+        significant bit of the result.
```

Comment:
So the essence of the example is you have x =  1 << 62, or 0100 followed by 60 zeroes.  So it's a 64bit number, which is the minimum required for AbstractInt.

If I compute (x << 1) then it's 1 followed by 63 zeroes.  That 1 bit in the most significant position is the sign bit, because numbers are twos complement.  So now the author expects ( x << 1 ) to have value 1 << 63.  But (x << 1) is negative.  So I call that overflow.

Overflow does *not* occur if and only if: all the discarded bits are the same value, and are the same as the resulting sign bit.    That is, when shifting by y bits, overflow does *not* occur if the most-significant y+1 bits are the same.

 I described the overflow case instead of the not-overflow case, and I wasn't certain which would read better.